### ISSUE TITLE
`astro/no-unsafe-inline-scripts` is erroneous

### ISSUE THREAD

**Issue by Lippiece** [ota-meshi/eslint-plugin-astro#538]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

v9.39.2

### What version of `eslint-plugin-astro` are you using?

1.5.0

### What did you do?

<details>
<summary>Configuration</summary>

```
Enable `astro/no-unsafe-inline-scripts`
```
</details>

```astro
<script>console.log("anything")</script>
```

### What did you expect to happen?

The rule should ignore scripts without `is:inline`.

### What actually happened?

The rule fires telling me the `<script>` is inline and as such unsafe. 

However, `<script>`s in Astro [get bundled](https://docs.astro.build/en/guides/client-side-scripts/) into the page's script and then imported. There's no CSP issue here.

### Link to Minimal Reproducible Example

I think there's nothing to debug, the rule is just erroneous, but I will submit reproduction if you insist.

### Additional comments

Script is inlined if you set `is:inline` yourself. In this case, the rule should report it.

**Comment by Lippiece** [comment]
Didn't notice the added label, sorry.

https://ota-meshi.github.io/eslint-plugin-astro/playground/#eJxFjEEOgjAQAL+y2ZMS1kaOgH7DSy+krNhk2Zp28UL4uxETzRwnMyuGNDK2SEReXQU3lpBmBkvARaIaPWWZotJQLCeonFevzoGwQYALnDuve/uhLyHHp129AoSkJQmfJE2H4dh57d3PYo15ES7Yrrh/nSYqbPSwWWiMmYPFF2Pb1H+/aBnuTFElKtP3VbBttu0NUpNDcA==
